File: need4spd/euler_14_other.rb
def chain(n)
	return 1 if n == 1
	n.even? ? chain(n/2)+1 : chain(3*n + 1)+1
end
p (1..1_000_000).map{ |n| [n, chain(n)] }.max{|a,b| a[1] <=> b[1]}

File: need4spd/euler_14_other.py
hseqdict = {} # {갯수: 숫자}
for i in range(1, 1000001):
    hseq = [i]
    while hseq[-1] != 1:
        if hseq[-1] % 2 == 0:
            hseq.append(hseq[-1] / 2)
        else:
            hseq.append(3 * hseq[-1] + 1)
    hseqdict[len(hseq)] = i

print (hseqdict[max(hseqdict.keys())])